src/TraceEvent/ETWTraceEventSource.cs

```diff
@@ -455,9 +455,9 @@ private void Initialize(string fileOrSessionName, TraceEventSourceType type)
                 sessionEndTimeQPC = this.UTCDateTimeToQPC(maxSessionEndTimeUTC);
             }
             Debug.Assert(_QPCFreq != 0);
-            if (pointerSize == 0)       // Real time does not set this (grrr). 
+            if (pointerSize == 0 || IsRealTime)  // We get on x64 OS 4 as pointer size which is wrong for realtime sessions. Fix it up. 
```

Comment:
Ideally we would not have the || IsRealTime in this condition.   Basically the ETW system in theory knows the pointer size and we should just be using that.  We only 'take control' and set it if it is clear that the ETW system did not (it set it to 0).   

Is it really the case that the pointerSize comes back as 4 on a when you create a real-time ETW source?